ibc-testkit/src/testapp/ibc/core/types.rs

```diff
@@ -469,39 +469,31 @@ where
     HostClientState<H>: ClientStateValidation<DefaultIbcStore>,
 {
     pub fn with_latest_height(height: Height) -> Self {
-        let context = TestContextConfig::builder()
+        let context: TestContext<_> = dummy_store_generic_test_context()
             .latest_height(height)
-            .build::<TestContext<_>>();
+            .build();
         LightClientBuilder::init().context(&context).build()
     }
 }
 
-#[derive(TypedBuilder)]
-#[builder(builder_method(name = init), build_method(into))]
-pub struct LightClientBuilder<'a, H>
-where
-    H: TestHost,
-    HostClientState<H>: ClientStateValidation<DefaultIbcStore>,
-{
-    context: &'a TestContext<H>,
-    #[builder(default, setter(into))]
-    consensus_heights: Vec<Height>,
-    #[builder(default)]
-    params: H::LightClientParams,
-}
+pub struct LightClientBuilder;
 
-impl<'a, H> From<LightClientBuilder<'a, H>> for LightClientState<H>
-where
-    H: TestHost,
-    HostClientState<H>: ClientStateValidation<DefaultIbcStore>,
-{
-    fn from(builder: LightClientBuilder<'a, H>) -> Self {
-        let LightClientBuilder {
-            context,
-            consensus_heights,
-            params,
-        } = builder;
+#[bon]
+impl LightClientBuilder {
+    #[builder(finish_fn = build)]
+    pub fn init<H>(
+        context: &TestContext<H>,
+
+        #[builder(default, into)] //
```

Comment:
Note that `rustfmt` does a bad job of formatting attributes on function arguments. Throughout the PR I tried to avoid using them (e.g. use `Option` type instead of adding `#[builder(default)]`), bit I used a workaround here with an empty line comment to nudge `rustfmt` into placing the attribute on a separate line before the function parameter (which reads better IMO).

I described this problem in the [docs about limitations](https://elastio.github.io/bon/guide/limitations#formatting-of-attributes-on-function-arguments)

Reply:
hey. Thanks for mentioning this. We will try it out at our end and decide. :raised_hands:

Reply:
I took a look at the examples from your doc and tried on our codebase. For now, I am ok with having `#[builder(default = ...)] arg: T` in the same line :slightly_smiling_face:

Reply:
Sure, I'll remove the workarounds